patches/mimalloc-v2.2.4/2_return_stat.patch

```diff
@@ -0,0 +1,66 @@
+diff --git a/src/alloc.c b/src/alloc.c
+index 893f3094..b54497ea 100644
+--- a/src/alloc.c
++++ b/src/alloc.c
+@@ -676,22 +676,49 @@ mi_decl_restrict void* _mi_heap_malloc_guarded(mi_heap_t* heap, size_t size, boo
+ }
+ #endif
+ 
+-bool mi_heap_page_is_underutilized(mi_heap_t* heap, void* p, float ratio) mi_attr_noexcept {
+-  mi_page_t* page = _mi_ptr_page(p);   // get the page that this belongs to
+ 
+-  mi_heap_t* page_heap = (mi_heap_t*)(mi_atomic_load_acquire(&(page)->xheap));
++mi_page_usage_stats_t mi_heap_page_is_underutilized(mi_heap_t *heap, void *p, float ratio,
++                                                    bool return_detailed_stats) mi_attr_noexcept {
++  mi_page_t *page = _mi_ptr_page(p); // get the page that this belongs to
++  mi_heap_t *page_heap = (mi_heap_t *) (mi_atomic_load_acquire(&(page)->xheap));
+ 
+-  // the heap id matches and it is not a full page
+-  if (mi_likely(page_heap == heap && page->flags.x.in_full == 0)) {
+-    // first in the list, meaning it's the head of page queue, thus being used for malloc
+-    if (page->prev == NULL)
+-      return false;
++  if (!return_detailed_stats) {
++    mi_page_usage_stats_t result = {.should_realloc = false};
++    if (mi_likely(page_heap == heap && page->flags.x.in_full == 0)) {
++      if (page->prev != NULL)
++        result.should_realloc = page->used <= (unsigned) (page->capacity * ratio);
++    }
++    return result;
++  }
++
++  mi_page_usage_stats_t result = {
++    .page_address = (uintptr_t) page,
++    .block_size = page->block_size,
++    .capacity = page->capacity,
++    .reserved = page->reserved,
++    .used = page->used,
++    .should_realloc = false,
++    .flags = 0,
++  };
++
++  if (page->flags.x.in_full == 1) {
++    result.flags |= 1;
```

Comment:
Please introduce defines around `mi_page_usage_stats_s` declaration for flags and add a comment around the flags mentioning these defines

Reply:
added